### ISSUE TITLE
Fork for Pale Moon

### ISSUE THREAD

**Issue by GJRobert** [hultmann/multifox#264]
Sorry again to see Multifox killed by e10s and the end of XUL/XPCOM.

I have used the new "container tab" of Firefox, which is definitely a nice feature and subsitute for Multifox. Nice innovation Mozilla.

The problem is, I, like quite a few users, have decided to say goodbye to Firefox due to the end of deep customization and XUL/XPCOM addons and turned to Pale Moon, which is faster, more power-efficient and more memory-efficient than current Firefox builds and which will keep the XUL addon ecology alive.

For the time being, [Multifox does not work on Pale Moon yet](https://forum.palemoon.org/viewtopic.php?t=9761), but personally I do hope a compatible fork emerge.

Just a public urge here, and thanks and honor to hultmann, who has built such a good tool of Multifox for these years.

**Comment by LimboSlam** [comment]
Yes, thanks again @hultmann for this wonderful and powerful add-on! :) 

However, I'm curios as to why you lost interest in developing your add-on. Especially when you have an independent browser (Pale Moon) that continues to support your technology. In fact, we have page just for add-on developers that are familiarizing themselves with our browser: https://developer.palemoon.org/Add-ons:Site. 

Anyways, this is totally your decision and I too hope a fork emerge.

**Comment by AWOL-TECH** [comment]
@hultmann 

I have been using multifox for years and the fact it is now broken has destroyed my world!!!! 

I haven't found any other plugins even close to what multifox does for me.

I really hope you decide to add support for things like palemoon or any other browsers, as I really don't want to carry on without it !

**Comment by SparkyKestrel** [comment]
@hultmann 

I would like to back other's requests to continue development for Multifox on Palemoon.

Multifox was a real boon for me being able open multiple Tumblr, Facebook etc accounts simultaneously without having multiple browsers open, and while Firefox's Container Tabs is not bad, the experience is nowhere near as good as Multifox (for example Container Tabs seems to get confused over having multiple Tumblr accounts open and always asks me to log in each time, no matter what profile I am using, whereas Multifox remembered all the different logins with no issues)
(and not to mention now with Firefox  dropping support for XPCOM extensions as of v57, which I am to understand will break almost every extension currently available, I have moved to Firefox 52 ESR for now, which Container Tabs does not work with)

If you were to start development of Multifox again for Palemoon I, and it seems many others would be able to move exclusively to Palemoon and say goodbye to Mozilla who seem more interested in screwing over their current userbase than continuing to support innovation &  difference in unique browser choice.

I myself am not a programmer/coder (I would have tried doing it myself if I was), so please consider making Multifox compatible with Palemoon, we would all be so grateful to you if were to do this.

**Comment by AWOL-TECH** [comment]
From what I have been reading @hultmann still very sadly has no intention of supporting webextentions etc, or porting multifox over to palemoon. 

@hultmann Seriously man I'll buy you a beer if you change your mind! 

ANYWAY. Has anyone found a GOOD alternative to multifox yet or even a fork or something?? Whether its via palemoon or firefox? 

Mozilla's container tabs still suck IMO. They don't fill the gap left by multifox. 
https://addons.mozilla.org/en-GB/firefox/addon/multi-account-containers 

I have even tried extending container tabs with - Containers on the go  and some other ones still nowhere near. 
https://addons.mozilla.org/en-GB/firefox/addon/containers-on-the-go/


Multifox does seem literally one of a kind ! I'm not extensions developer so Its not something I can do myself, However I do build websites and would be happy to help set up a site for a new project or something if anyone is doing it. Or contributing in any other way

With Novembers v57approaching and the realisation that more than half of my current extensions are going to stop working (Aside from the huge plugins, many devs seem to be jumping ship and not updating) my firefox is going to stop working and my workflow is going to be severely hindered (using this in a work environment). Mozilla seem to have an endless amount of feature requests for their container tabs so theres absolutely no point request multifox features as it will most likely take years. 

I could even settle for a plugin that just sets different sessions across browser windows rather than tabs, for example I need window 1 to login google account 1 and window 2 to login to google account 2 etc but it must persist and allow dual sessions to places like google. (this is a very basic example, realistically it would be 5 different sessions for 5 windows {with each of the windows setting all its tabs to the same session) then the other windows would remain default session)

sorry for reposting an old thread but trying to find the answers is becoming a very difficult task

**Comment by LimboSlam** [comment]
Moonchild (main developer and founder) has recently updated the guidelines on creating an exclusive/hybrid extension for Pale Moon/Firefox: https://forum.palemoon.org/viewtopic.php?f=19&t=16785&p=122637#p122637. I would take a look at their repository of extensions, including add-ons and complete themes: https://addons.palemoon.org/. 

We are gaining more contributors, and many developers are migrating to Pale Moon as their main browser or add-on development. You may also be interested in knowing their future plans: https://www.palemoon.org/roadmap.shtml.

**Comment by GJRobert** [comment]
@timeassistant Hi Greg,

Here is my alternative combo pack to gain some portion of Multifox capability:
1. First of all, I'm using Pale Moon now, which supports legacy addons far better.
1. Install CookieSwap.
1. Duplicate your profile and set up them as secondary/tertiary profiles.
1. Install ProfileSwitcher in your main swiss knife.

Now, open secondary/tertiary profiles as needed, and use CookieSwap in those, to toggle to other identities.

This is far inefficient compared to Multifox and even Firefox official Container (2017), but at least... there is something usable on Pale Moon now.

**Comment by hultmann** [comment]
Sorry folks, I moved on to other projects, no interest in resuming Multifox development

**Comment by AWOL-TECH** [comment]
@LimboSlam  @GJRobert  Thanks both for your info there, I have looked into Pale Moon previously, but sadly I never did find a good solution. However Roberts workaround sounds like it could be worth a shot. 

In all honesty I will need to do some serious testing before considering a switch but this is definitely progress, thanks guys,


@hultmann sorry to hear it, but I understand. Best of luck with your future projects.

**Comment by LimboSlam** [comment]
Hey @hultmann, no worries! It was a pleasure using Multifox. Thanks again! 

And yes, @GJRobert solution is currently the best workaround. Maybe @JustOff, @FranklinDM, @janekptacijarabaci or @wsdfhjxc could fork a compatible version of the add-on, and bring it up to par with Pale Moon? Hopefully, the license allows it.

**Comment by janekptacijarabaci** [comment]
I'm sorry, but I do not plan to do another fork in the future.

**Comment by fofajardo** [comment]
I'll check it first if it's feasible and wouldn't be too much of a burden to maintain.
_EDIT: I might convert this to become a classic-style extension if I have time._

**Comment by leedoyle** [comment]
@GJRobert please elaborate on how to reproduce it. I didn't understand how you managed to combine ProfileSwitcher and CookieSwap. 
Is CookieSwap able to switch between different profiles? But why using ProfileSwitcher, then?
Please explain.

**Comment by endolith** [comment]
Also Waterfox still supports XUL, but multifox doesn't work in it?

**Comment by LimboSlam** [comment]
:sunglasses:

**Comment by GJRobert** [comment]
@endolith I have tried Multifox 3.2.3 on Pale Moon (yes we're talking about PM, not Waterfox), but the toolbar button does not appear.

Multifox 2.0.7 seems to be working on a new PM profile, but this old version does not support the latest Gmail HTTP Basic Authentication.

So for Multifox to be fully compatible on PM, an update is needed.

**Comment by GJRobert** [comment]
@leedoyle CookieSwap can swap cookies and switch to another identity instantly. But the problem is you cannot use different identities **SIMULTANEOUSLY** (unlike Multifox with the functionality to open different identifies in the same profile at the same time). That's why I'd like to use ProfileSwitcher so to open multiple Fx/PM instances, and using different identities each with CookieSwap.

I was sorry that I did not make this clear. But I was quite busy and your tone seemed a bit not so welcoming, so I haven't post my accounting since.

**Comment by ackern** [comment]
Is Multifox working for anyone, at all, even partially, in any way, on Pale Moon? For me Pale Moon (28.7.2) is able to add Multifox 3.2.3 and it shows up in the list of extensions but I don't see any way to use it. No toolbar button or menu item anywhere.  I mostly just need it for Facebook.

Also FYI there is a containers extension in Classic Add-Ons Archive called Firefox Multi-Account Containers but I have the same problem with it - it installs but nothing happens, no button or menu item.

**Comment by GJRobert** [comment]
> Is Multifox working for anyone, at all, even partially, in any way, on Pale Moon? For me Pale Moon (28.7.2) is able to add Multifox 3.2.3 and it shows up in the list of extensions but I don't see any way to use it. No toolbar button or menu item anywhere. I mostly just need it for Facebook.
> 
> Also FYI there is a containers extension in Classic Add-Ons Archive called Firefox Multi-Account Containers but I have the same problem with it - it installs but nothing happens, no button or menu item.

@ackern If you are using Pale Moon for compatibility with XUL/XPCOM addons, perhaps you can try Waterfox Classic, which I am using since 2018, which is based on Firefox 56 with XUL addon compatibility, and also with Firefox multi-account container feature built in.
